lib/datasource/docker/common.ts

```diff
@@ -511,3 +511,14 @@ export async function getLabels(
     return {};
   }
 }
+
+export function isECRMaxResultsError(err): boolean {
+  return (
+    err.response?.statusCode === 405 &&
+    err.response?.headers['docker-distribution-api-version'] &&
+    // https://docs.aws.amazon.com/AmazonECR/latest/APIReference/API_DescribeRepositories.html#ECR-DescribeRepositories-request-maxResults
+    err.response.body?.errors[0]?.message?.includes(
```

Comment:
```suggestion
    err.response.body?.errors.?[0]?.message?.includes(
```
What if errors don't exist?

Reply:
Nice catch! I also added a guard for the headers being present. Tests to add confidence, changes here https://github.com/renovatebot/renovate/pull/13531/commits/85810340faf23024ed5cc38c199f952cc6266f6c.